weed/shell/command_volume_fsck.go

```diff
@@ -376,12 +379,34 @@ func (c *commandVolumeFsck) collectOneVolumeFileIds(tempFolder string, dataNodeI
 		if err != nil {
 			return fmt.Errorf("failed to start copying volume %d%s: %v", volumeId, ext, err)
 		}
-
-		err = writeToFile(copyFileClient, getVolumeFileIdFile(tempFolder, dataNodeId, volumeId))
+		var buf bytes.Buffer
+		for {
+			resp, err := copyFileClient.Recv()
+			if errors.Is(err, io.EOF) {
+				break
+			}
+			if err != nil {
+				return err
+			}
+			buf.Write(resp.FileContent)
+		}
+		index_size := len(buf.Bytes())
+		index, err := idx.FirstLargerIndex(buf.Bytes(), index_size, func(key types.NeedleId, offset types.Offset, size types.Size) (bool, error) {
+			resp, err := volumeServerClient.VolumeNeedleStatus(context.Background(), &volume_server_pb.VolumeNeedleStatusRequest{
+				VolumeId: volumeId,
+				NeedleId: uint64(key),
```

Comment:
If one key is updated multiple times, this would read the needle status for the last entry, not the status at the specific offset.